pre-commit.sh

```diff
@@ -60,7 +60,7 @@ echo -e "$(GREEN INFO): cargo clippy ..."
 
 # Cargo clippy always return exit code 0, and `tee` doesn't work.
 # So let's just run cargo clippy.
-cargo clippy
+cargo clippy --all-targets --workspace --features avro,pyarrow -- -D warnings
```

Comment:
I wonder if it would be better to run the same exact script CI does, namely

https://github.com/apache/arrow-datafusion/blob/main/ci/scripts/rust_clippy.sh

(the CI runs it here https://github.com/apache/arrow-datafusion/blob/e86253919983ffe89d74ec1310f5f080482adcff/.github/workflows/rust.yml#L475C14-L475C39)

Reply:
I tried to just use `source ./ci/scripts/rust_clippy.sh` here, but with `set -ex` in it, the process would terminate if clippy is not happy which causes subsequent `fmt` won't execute.
So I copy the whole cargo command.